rcl/src/rcl/client.c

```diff
@@ -113,7 +109,29 @@ rcl_client_init(
     RCL_SET_ERROR_MSG(rmw_get_error_string().str);
     goto fail;
   }
-
+  
+  client->impl->service_event_publisher = NULL;
+  if (options->enable_service_introspection) {
+    client->impl->service_event_publisher = allocator->zero_allocate(
```

Comment:
It doesn't seem like a zero allocation is necessary since we're setting the memory to a zero initialized object anyways at L125. I would just make a regular allocate call.

Reply:
Fixed in 2f1e9fc252c60380f4157bbb01266f3ce87c2863.